src/Model/Behavior/UploadBehavior.php

```diff
@@ -224,26 +224,29 @@ public function getWriter(EntityInterface $entity, ?UploadedFileInterface $data
      * @param string $field the field for which data will be saved
      * @param array $settings the settings for the current field
      * @param string $basepath a basepath where the files are written to
+     * @param string $filename Filename from processor to use
+     *
      * @return array key/value pairs of temp files mapping to their names
      */
     public function constructFiles(
         EntityInterface $entity,
         UploadedFileInterface $data,
         string $field,
         array $settings,
-        string $basepath
+        string $basepath,
```

Comment:
Instead of having 2 arguments for basepath and filename it's better to just pass the path transformer instance

Reply:
That doesn't work, it calls the nameCallback multiple times resulting in inconsistent file names

Reply:
Line #124 `$entity->set($field, $filename);` expects the filename. Since there's no way for Line #119 `$success = $writer->write($files);` to return the transformed filename, we must pass it to the transformer so it can map it correctly. Otherwise Line #114 is stale and out of date and results in a filename that is out of sync with what is stored in the FlySystem.